packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/components/ReminderForm/OverdueReminderForm.tsx

```diff
@@ -24,114 +23,167 @@ import {
   Stack,
   Typography,
   InputLabel,
+  Alert,
 } from '@mui/material';
 
 import { ReminderFormLayout } from './ReminderFormLayout';
 import { getOverdueValidationSchema } from './validation';
 
-interface CreateOverdueReminderProps {
-  id?: string;
+interface OverdueReminderBasePropsProps {
+  onClose?(): void;
 }
 
-export const CreateOverdueReminder = ({ id }: CreateOverdueReminderProps) => {
-  const { i18n } = useLingui();
-  const dialogContext = useDialog();
-  const { api, queryClient } = useMoniteContext();
+interface CreateOverdueReminderProps extends OverdueReminderBasePropsProps {
+  onCreate?(reminderId: string): void;
+  reminderId?: never;
+}
 
+interface UpdateOverdueReminderProps extends OverdueReminderBasePropsProps {
+  reminderId: string;
+  onUpdate?(reminderId: string): void;
+}
+
+export const OverdueReminderForm = ({
+  reminderId,
+  ...restProps
+}: CreateOverdueReminderProps | UpdateOverdueReminderProps) => {
+  const { i18n } = useLingui();
+  const { api } = useMoniteContext();
   const {
-    data: overdueReminder,
-    error: overdueReminderQueryError,
+    data: reminder,
+    error,
+    isError,
     isLoading,
   } = api.overdueReminders.getOverdueRemindersId.useQuery(
     {
       path: {
-        overdue_reminder_id: id || '',
+        overdue_reminder_id: reminderId || '',
       },
     },
     {
-      enabled: Boolean(id),
+      enabled: Boolean(reminderId),
     }
   );
 
-  const methods = useForm({
+  if (isLoading) {
+    return <LoadingPage />;
+  }
+
+  if (isError) {
+    return (
+      <NotFound
+        title={t(i18n)`Overdue reminder error`}
+        description={getAPIErrorMessage(i18n, error)}
+      />
+    );
+  }
+
+  if (reminderId && !reminder) {
+    return (
+      <NotFound
+        title={t(i18n)`Reminder not found`}
+        description={t(
+          i18n
+        )`There is no reminder by provided id: ${reminderId}`}
+      />
+    );
+  }
+
+  return <CreateOverdueReminderComponent {...restProps} reminder={reminder} />;
+};
+
+const CreateOverdueReminderComponent = ({
+  reminder,
+  ...props
+}: (
+  | {
+      onCreate?(reminderId: string): void;
+    }
+  | { onUpdate?(reminderId: string): void }
+) &
+  OverdueReminderBasePropsProps & {
+    reminder?: components['schemas']['OverdueReminderResponse'];
```

Comment:
```ts
}: (
  | {
      onCreate?(reminderId: string): void;
    }
  | { onUpdate?(reminderId: string): void }
) &
  OverdueReminderBasePropsProps & {
    reminder?: components['schemas']['OverdueReminderResponse'];
  }
```

I would prefer to move this in a separate interface as the amount of types here is quite large

Reply:
I am really curious, why is this component not even exported?

Reply:
Not sure, I follow your question. I'm asking about types and moving the as separate interface